framework/src/service/ServiceReferenceBasePrivate.cpp

```diff
@@ -119,41 +122,83 @@ std::shared_ptr<void> ServiceReferenceBasePrivate::GetService(BundlePrivate* bun
 
 InterfaceMapConstPtr ServiceReferenceBasePrivate::GetServiceInterfaceMap(BundlePrivate* bundle)
 {
+  /*
```

Comment:
Maybe I have missed it in the spec. Do you mind pointing me to the right section?

I found the snippet "If this method is recursively called for the specified bundle, a
framework event of type FrameworkEvent.ERROR is fired containing a service exception of type ServiceException.FACTORY_RECURSION and null is returned to the bundle." on [page#255 of the OSGi core spec](https://osgi.org/download/r6/osgi.core-6.0.0.pdf#page=255). It does not call out recursion on the same thread. 

Also, I was thinking when we add code to satisfy a requirement from the spec, it would be helpful if we had a comment with the link to spec page where the requirement came from. Would be easier to track in the future.